app/src/main/java/com/hercan/weatherapp2/uı/detail/DetailActivity.kt

```diff
@@ -0,0 +1,47 @@
+package com.hercan.weatherapp2.uı.detail
+
+import android.os.Build
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.hercan.weatherapp2.R
+import com.hercan.weatherapp2.databinding.ActivityDetailBinding
+import com.hercan.weatherapp2.presentation.model.CityWeatherItem
+import com.hercan.weatherapp2.presentation.model.getTemperature
+import com.hercan.weatherapp2.presentation.viewbinding.viewBinding
+
+class DetailActivity : AppCompatActivity() {
+
+    private val binding by viewBinding(ActivityDetailBinding::inflate)
+    private var cityWeatherItem: CityWeatherItem? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(binding.root)
+
+        getData()
+        bindUI(cityWeatherItem)
+    }
+
+    private fun getData() {
+        cityWeatherItem = if (Build.VERSION.SDK_INT >= 33) {
+            intent.getParcelableExtra("cityWeatherItem", CityWeatherItem::class.java)
```

Comment:
```suggestion
            intent.getParcelableExtra("cityWeatherItem", CityWeatherItem::class.java)
```
`cityWeatherItem` bu string'i `companion object` icerisinde `const val` olarak tutabiliriz.